src/hid.c

```diff
@@ -148,6 +148,36 @@ fido_dev_info_ptr(const fido_dev_info_t *devlist, size_t i)
 	return (&devlist[i]);
 }
 
+int
+fido_dev_info_set(fido_dev_info_t *devlist, size_t i,
+    const char *path, const char *manufacturer, const char *product,
+    const fido_dev_io_t *io, const fido_dev_transport_t *transport)
+{
+	char *path_copy = NULL, *manu_copy = NULL, *prod_copy = NULL;
+	int r = FIDO_ERR_INTERNAL;
+
+	if ((path && (path_copy = strdup(path)) == NULL) ||
+	    (manufacturer && (manu_copy = strdup(manufacturer)) == NULL) ||
+	    (product && (prod_copy = strdup(product)) == NULL))
+		goto out;
+
+	devlist[i].path = path_copy;
+	devlist[i].manufacturer = manu_copy;
+	devlist[i].product = prod_copy;
+	if (io)
+		devlist[i].io = *io;
+	if (transport)
+		devlist[i].transport = *transport;
+	r = FIDO_OK;
+
+out:	if (r) {
```

Comment:
Prefer having the label on its own line and checking the return value against `FIDO_OK` explicitly.